backend/layers/api/portal_api.py

```diff
@@ -329,15 +334,25 @@ def get_collection_index(self):
 
     def delete_collection(self, collection_id: str, token_info: dict):
         """
-        Deletes a collection version from the persistence store.
+        Deletes a collection version from the persistence store, or tombstones a canonical collection.
         """
-        version = self.business_logic.get_collection_version(CollectionVersionId(collection_id))
+        resource_id = CollectionVersionId(collection_id)
+        version = self.business_logic.get_collection_version(resource_id)
         if version is None:
-            raise ForbiddenHTTPException()
+            resource_id = CollectionId(collection_id)
+            version = self.business_logic.get_collection_version_from_canonical(resource_id)
+            if version is None:
+                raise ForbiddenHTTPException()
         if not UserInfo(token_info).is_user_owner_or_allowed(version.owner):
             raise ForbiddenHTTPException()
 
-        self.business_logic.delete_collection_version(CollectionVersionId(collection_id))
+        if isinstance(resource_id, CollectionVersionId):
```

Comment:
I am not sure about using `isinstance` in this context. That said, I can't come up with a better solution myself.

Reply:
we can replace with a flag if you prefer. 

Or we can rework the underlying business logic method to fetch the collection/collection version before deleting and handle the logic there. The only issue is we'd be introducing a 2nd set of lookups, since we need the lookup here to get the version owner for the auth logic handling

Reply:
Let's leave it as is